From: Jiang Xin <jiangxin@ossxp.com>
Subject: [PATCH] t/wildrepo_detect

Smart wildrepo name detect, or in some cases wildrepo
may created automatically as normal repository.

Signed-off-by: Jiang Xin <jiangxin@ossxp.com>

---
 src/gitolite_rc.pm  |    2 +-
 src/gl-compile-conf |    1 +
 2 个文件被修改，插入 2 行(+)，删除 1 行(-)

diff --git a/src/gitolite_rc.pm b/src/gitolite_rc.pm
index 9f65a7d..f9818a1 100644
--- a/src/gitolite_rc.pm
+++ b/src/gitolite_rc.pm
@@ -48,7 +48,7 @@ $W_COMMANDS=qr/^git[ -]receive-pack$/;
 $REPONAME_PATT=qr(^\@?[0-9a-zA-Z][0-9a-zA-Z._\@/+-]*$);
 $USERNAME_PATT=qr(^\@?[0-9a-zA-Z][0-9a-zA-Z._\@+-]*$);
 # same as REPONAME, but used for wildcard repos, allows some common regex metas
-$REPOPATT_PATT=qr(^\@?[0-9a-zA-Z[][\\^.$|()[\]*+?{}0-9a-zA-Z._\@/,-]*$);
+$REPOPATT_PATT=qr(^(\^|\@)?[0-9a-zA-Z[][\\^.$|()[\]*+?{}0-9a-zA-Z._\@/,-]*$);
 # pattern for refnames pushed or names of files changed
 $GL_REF_OR_FILENAME_PATT=qr(^[0-9a-zA-Z][0-9a-zA-Z._\@/+ :,-]*$);
 
diff --git a/src/gl-compile-conf b/src/gl-compile-conf
index f497ae5..30ab732 100755
--- a/src/gl-compile-conf
+++ b/src/gl-compile-conf
@@ -435,6 +435,7 @@ sub do_normal_repos
     for my $repo (sort @repos) {
         next unless $repo =~ $REPONAME_PATT;    # skip repo patterns
         next if $repo =~ m(^\@|EXTCMD/);        # skip groups and fake repos
+        next if $GL_WILDREPOS and ( $repo =~ /(^\^|\$$)/ or $repo =~ /\.(\*|\+)/ ); # is wild repos
 
         unless ($GL_NO_CREATE_REPOS) {
             unless (-d "$repo.git") {
-- 
tg: (f73e3d7..) t/wildrepo_detect (depends on: upstream)
